posts/inside-rust/2019-10-11-AsyncAwait-Not-Send-Error-Improvements.md

```diff
@@ -121,9 +121,25 @@ async fn bar(x: &Mutex<u32>) {
 ```
 
 ```rust
-enum BarGenerator {
-    Unresumed { x: &Mutex<u32>, g: MutexGuard<'_, u32>, },
-    Suspend0 { x: &Mutex<u32>, g: MutexGuard<'_, u32>, }
+enum BarAsyncFnGenerator {
+    Unresumed {
+        // async fn params
+        x: &Mutex<u32>,
+    },
+
+    Suspend0 {
+        // future being polled: baz_fut = baz()
+        baz_fut: BazAsyncFnGenerator,
+
+        // locals that cross the await point
+        x: &Mutex<u32>,
+        g: MutexGuard<'_, u32>,
+    },
+
+    Return(
```

Comment:
I'd prefer this be a unit variant named `Returned`. This follows the convention followed by Tyler in [his blog post](https://tmandry.gitlab.io/blog/posts/optimizing-await-1/#layout-variants) that I link in the paragraph below.